pkg/controller/util/safe_map.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"sync"
+)
+
+type SafeMap struct {
```

Comment:
Is there any particular reason we could not use thread-safe concurrent map type from golang library https://golang.org/pkg/sync/#Map

Reply:
Because of the recommendations there.. :-). I had a look at the implementation too and found it doing lot of additional stuff then that was needed here, especially the `Range()` method (which I could accomplish for now implementing `GetAll()` here).